Document inclusion of package.json in workspace task cache keys (#1955)
vercel/turborepo: docs/pages/docs/core-concepts/caching.mdx

```

<Callout type="info">
  `package.json` is *always* considered an input for tasks in the workspace it lives in.

>   This is because the *definition* of the task itself lives in package.json in the `scripts` key.

Reviewer: ```suggestion
  This is because the *definition* of the task itself lives in `package.json` in the `scripts` key.
```

  If you change that, any cached output is considered invalid.
</Callout>

If you want _all_ tasks to depend on certain files, you can declare this dependency in the
`globalDependencies` array.

```diff
{
  "$schema": "https://turborepo.org/schema.json",
+  "globalDependencies": [".env"],
  "pipeline": {
    // ...other tasks
    "test": {
      "outputs": [], // leave empty to only cache logs
      "dependsOn": ["build"],
      "inputs": ["src/**/*.tsx", "src/**/*.ts", "test/**/*.ts"]
    }
  }
}
```

<Callout type="info">

>   `turbo.json` is *always* considered a global input. If you modify `turbo.json`, all caches

Reviewer: ```suggestion
  `turbo.json` is *always* considered a global dependency. If you modify `turbo.json`, all caches
```

  are invalidated.
</Callout>

## Altering Caching Based on Environment Variables

When you use `turbo` with tools that inline environment variables at build time
(e.g. [Next.js][1] or [Create React App][2]), it is important to tell `turbo` about it.
Otherwise, you could ship a cached build with the wrong environment variables!

> Luckily, you can control `turbo`'s caching behavior (also known as "hashing") based on

Reviewer: ```suggestion
You can control `turbo`'s caching behavior based on
```

@mattpocock just did a pass removing this sort of "how it works" stuff because it doesn't _really_ matter.

Reviewer 2: 👍

      ],
      "outputs": [".next/**"],
    },
  }
}
```

To alter the cache for _all_ tasks, you can declare environment variables in the

> `globlEnv` array:

Author: ```suggestion
`globalEnv` array:
```

> `globalDependencies` array:

Author: ```suggestion
`globlEnv` array:
```

```diff
{
  "$schema": "https://turborepo.org/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": [
        "^build",
        // env vars will impact hashes of all "build" tasks
        "$SOME_ENV_VAR",
      ],
      "outputs": ["dist/**"]
    },

    // override settings for the "build" task for the "web" app
    "web#build": {
      "dependsOn": [
        "^build",
        // env vars that will impact the hash of "build" task for only "web" app
        "$STRIPE_SECRET_KEY",
        "$NEXT_PUBLIC_STRIPE_PUBLIC_KEY",
        "$NEXT_PUBLIC_ANALYTICS_ID",
      ],
      "outputs": [".next/**"],
    },
  },

> +  "globalDependencies": [
> +    "$GITHUB_TOKEN",// env var that will impact the hashes of all tasks,
> +    "tsconfig.json", // file contents will impact the hashes of all tasks,
> +    ".env.*", // glob file contents will impact the hashes of all tasks,
> +  ]

Author: ```suggestion
+  // env var that will impact the hashes of all tasks,
+  "globalEnv": [ "GITHUB_TOKEN"] 
```
